libsolidity/interface/CompilerStack.cpp

```diff
@@ -1206,13 +1206,69 @@ Json CompilerStack::ethdebug(Contract const& _contract, bool _runtime) const
 	solAssert(m_stackState >= AnalysisSuccessful, "Analysis was not successful.");
 	solAssert(_contract.contract);
 	solUnimplementedAssert(!isExperimentalSolidity());
-	if (_runtime)
+	Json result = {
+	{"contract",
+		{
+			{"name", _contract.contract->name()},
+			{"definition",
+				{{"sources",
+					{{"id", static_cast<int>(sourceIndices()[_contract.contract->sourceUnitName()])}}
+				}}
+			}}
+		},
+		{"environment ", _runtime ? "call" : "create"},
+		{"instructions", ethdebugInstructions(_contract, _runtime)}
+	};
+	return result;
+}
+
+Json CompilerStack::ethdebugInstructions(Contract const& _contract, bool _runtime) const
+{
+	evmasm::LinkerObject const& object = _runtime ? _contract.runtimeObject : _contract.object;
+	std::shared_ptr<evmasm::Assembly> const& assembly = _runtime ? _contract.evmRuntimeAssembly : _contract.evmAssembly;
+
+	if (assembly)
 	{
-		Json result = Json::object();
-		return result;
+		solUnimplementedAssert(
+			assembly->eofVersion() == std::nullopt && assembly->codeSections().size() == 1,
+			"ethdebug does not yet support EOF.");
```

Comment:
```suggestion
		solUnimplementedAssert(
			assembly->eofVersion() == std::nullopt,
			"ethdebug does not yet support EOF."
		);
		solAssert(assembly->codeSections().size() == 1);
```

I think it would be cleaner to just test the `eofVersion` and in a second assert then assert that there is only one code section.